src/gourmet/gtk_extras/dialog_extras.py

```diff
@@ -917,14 +917,11 @@ def saveas_file(title: str,
                              buttons=buttons,
                              show_filetype=show_filetype,
                              parent=parent)
-    # Get the filename, and the extension of the chosen filetype menu
-    filename, *extension = sfd.run()
-    extension = ''.join(extension)
-
-    # TODO: it would be cleaner to return the extension, rather than the
-    # output file type description
-    export_type = sfd.ext_to_filter[extension].get_name()
-    return filename, export_type
+    try:
+        filename, export_type = sfd.run()
```

Comment:
The list here is explicitly unpacked to catch empty lists (if the export was cancelled) to correctly return `None`s to the export manager, cleanly cancelling the action.